### PR TITLE
DRIVERS-2033 Require hello command + OP_MSG when 'loadBalanced=True'

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by blink1073 — DISMISSED** [review]
Do we need a new spec source demonstrating this behavior?

**Comment by juliusgeo** [comment]
@blink1073 See [the previous OP_MSG ticket](https://jira.mongodb.org/browse/DRIVERS-1873?filter=-1)

"Investigate whether it is practical to specify a test to validate this behavior. Failpoints cannot distinguish wire protocol versions. There may be no way to practically test this until SERVER-59085 is complete. After SERVER-59085 is complete, this will be tested in existing driver tests against the latest built servers (by virtue of authentication succeeding)."

**Review by blink1073 — APPROVED** [review]
LGTM!

File: source/mongodb-handshake/handshake.rst [hunk 1]
```diff
@@ -76,13 +76,12 @@ always be sent using the ``OP_MSG`` protocol. ``isMaster`` is referred to as
...
 for the initial handshake and use the ``OP_MSG`` protocol. If server API
-version is not requested drivers MUST
-use legacy hello for the first message of the initial handshake with the
-``OP_QUERY`` protocol (before switching to ``OP_MSG`` if the
-``maxWireVersion`` indicates compatibility), and include ``helloOk:true``in
-the handshake request.
+version is not requested and ``loadBalanced: False``, drivers MUST use legacy
+hello for the first message of the initial handshake with the ``OP_QUERY``
+protocol (before switching to ``OP_MSG`` if the ``maxWireVersion`` indicates
+compatibility), and include ``helloOk:true``in the handshake request.
```
Comment (ShaneHarvey) [inline source/mongodb-handshake/handshake.rst:84]: There's a missing space:
```
``helloOk:true``in
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
@juliusgeo I think this change warrants being implemented in pymongo before we merge. I'd like to make sure that the Serverless tests still pass.

**Review by ShaneHarvey — APPROVED** [review]
LGTM, I suggest renaming the PR title and the commit message to include the "hello" command requirement to be a little more informative. For example, "DRIVERS-2033 Require hello command + OP_MSG when 'loadBalanced=True'"

**Comment by juliusgeo** [comment]
As per @ShaneHarvey's about serverless tests passing in PyMongo with this change:
- [Evergreen build with serverless, load balancer and versioned api tests passing](https://evergreen.mongodb.com/version/61ef62df0ae6062a0ba30101?redirect_spruce_users=true) and [the corresponding commit](https://github.com/mongodb/mongo-python-driver/pull/837/commits/6587126930b8c502fc1f49241082d7b24d790584)
- [The PR](https://github.com/mongodb/mongo-python-driver/pull/837)

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Review by durran — APPROVED** [review]
LGTM

**Pull request merged**